GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Client/TestRailClientWrapper.cs

```diff
@@ -90,17 +127,56 @@ private static void ValidateRequestResult<T>(RequestResult<T> requestResult)
 
         public IEnumerable<Section> GetSections(ulong projectId)
         {
-            var result = _testRailClient.GetSections(projectId);
+            var policy = CreatePolicy<IList<Section>>();
+            var result = policy.Execute(()=>_testRailClient.GetSections(projectId));
             ValidateRequestResult(result);
             return result.Payload;
-            
         }
 
         public IEnumerable<Case> GetCases(ulong projectId, ulong suiteId)
         {
-            var result = _testRailClient.GetCases(projectId, suiteId);
+            var policy = CreatePolicy<IList<Case>>();
+            var result = policy.Execute(()=> _testRailClient.GetCases(projectId, suiteId));
             ValidateRequestResult(result);
             return result.Payload;
         }
+
+        /// <summary>
+        /// Pauses sending of requests when requests per minutes limit is reached 
+        /// </summary>
+        /// <param name="elapsedMilliSeconds">Milliseconds elapsed from start</param>
+        public void RequestsLimitCheck(double elapsedMilliSeconds)
+        {
+            var maxRequests = _config.TestRailMaxRequestsPerMinute;
+            //To ensure that limiter works even when one or more minute has passed
+            elapsedMilliSeconds %= 60_000;
+            if (RequestsCount + 1 >= maxRequests &&
+                elapsedMilliSeconds <= 59_000)
+            {
+                //additional 3000 milliseconds of sleep - just in case
+                var sleepTime = (int) Math.Round(63_000 - elapsedMilliSeconds);
+                Log.Debug($"Limit of {_config.TestRailMaxRequestsPerMinute} requests per minute is reached. Waiting for {sleepTime} seconds to continue...");
+                Thread.Sleep(sleepTime);
+                RequestsCount = 0;
+                Log.Debug($"Waiting completed. Requests count set to 0");
+            }
+            else if (RequestsCount >=_config.TestRailMaxRequestsPerMinute) 
+                RequestsCount = 0;
+        }
+
+        /// <summary>
+        /// RetryPolicy for request result of given type 
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        private RetryPolicy<RequestResult<T>> CreatePolicy<T>()
```

Comment:
Let's rename as `CreateSuccesStatusCodePolicy`?

Reply:
Maybe CreateResultHandlerPolicy will be better?